### PR TITLE
ARROW-2179: [C++] Install omitted headers in arrow/util

### PR DESCRIPTION
**Pull request by wesm**

### CODE REVIEW

File: cpp/src/arrow/util/CMakeLists.txt [hunk 1]
```diff
@@ -33,15 +33,17 @@ install(FILES
   compression_zstd.h
   cpu-info.h
   decimal.h
-  hash-util.h
   hash.h
+  hash-util.h
+  io-util.h
```
Comment (cpcloud) [inline cpp/src/arrow/util/CMakeLists.txt]: Looks like `memory.h` is also missing

Comment (cpcloud) [reply cpp/src/arrow/util/CMakeLists.txt]: Can we just glob files to avoid having to update this list?

Comment (wesm) [reply cpp/src/arrow/util/CMakeLists.txt]: Yeah, that would be better, taking a look

**Comment by wesm** [comment]
Done. This solution won't work for other directories which have some headers which should not be installed. See https://issues.apache.org/jira/browse/ARROW-2183

**Review by cpcloud — APPROVED** [review]
Nice. +1

**Pull request closed without merging**